packages/plugin-spatial-nback/src/index.ts

```diff
@@ -0,0 +1,509 @@
+      ">`;
+      
+      for (let row = 0; row < trial.rows; row++) {
+        html += '<div style="display: flex;">';
+        for (let col = 0; col < trial.cols; col++) {
+          html += `<div id="cell-${row}-${col}" style="
+            width: ${final_cell_size}px;
+            height: ${final_cell_size}px;
+            border: 1px solid #ccc;
+            background-color: white;
+            box-sizing: border-box;
+          "></div>`;
+        }
+        html += '</div>';
+      }
+      html += '</div></div>'; // Close grid and center container
+
+      // Button and feedback at bottom
+      html += `<div style="
+        flex: 0 0 auto;
+        display: flex;
+        flex-direction: column;
+        align-items: center;
+        gap: 0.75vh;
+        margin-bottom: clamp(12px, 4vmin, 24px);
+      ">`;
+      
+      // Feedback text first (directly under grid)
+      html += `<div id="nback-feedback" style="
+        height: 40px;
+        padding: clamp(6px, 1vmin, 24px);
+        font-size: clamp(14px, 2vmin, 20px);
+        font-weight: bold;
+        text-align: center;
+        width: 100%;
+      "></div>`;
+      
+      // Button second (under feedback text)
+      html += `<button id="nback-response-btn" style="
+        font-size: clamp(18px, 3vmin, 26px);
+        padding: clamp(18px, 4vmin, 40px) clamp(35px, 5vmin, 60px);
+        background-color: #2196F3;
+        color: white;
+        border: none;
+        border-radius: 8px;
+        cursor: pointer;
+        font-weight: bold;
+        box-shadow: 0 4px 8px rgba(0,0,0,0.2);
+        transition: all 0.2s;
+      " disabled>${trial.button_text}</button>`;
+      
+      html += '</div></div>'; // Close bottom container and main container
+      
+      display_element.innerHTML = html;
+
+      // Add button hover effects and event listener
+      const button = document.getElementById('nback-response-btn') as HTMLButtonElement;
+      button.addEventListener('mouseenter', () => {
+        if (!button.disabled) {
+          button.style.backgroundColor = '#1976D2';
+          button.style.transform = 'translateY(-2px)';
+        }
+      });
+      button.addEventListener('mouseleave', () => {
+        button.style.backgroundColor = '#2196F3';
+        button.style.transform = 'translateY(0)';
+      });
+      button.addEventListener('click', handleResponse);
+    };
+
+    const startTrial = (): void => {
+      // Show stimulus
+      const cell = document.getElementById(`cell-${stimulus_row}-${stimulus_col}`) as HTMLElement;
+      cell.style.backgroundColor = trial.stimulus_color;
+
+      // Enable response
+      response_allowed = true;
+      trial_start_time = performance.now();
+      stimulus_hidden = false;
+      
+      const responseButton = document.getElementById('nback-response-btn') as HTMLButtonElement;
+      responseButton.disabled = false;
+
+      // Set timeout to hide stimulus
+      stimulus_timeout = window.setTimeout(() => {
```

Comment:
use jsPsych.pluginAPI.setTimeout() which allows for jsPsych.pluginAPI.clearAllTimeouts()